.circleci/config.yml

```
@@ -208,9 +208,44 @@ commands:
 
   run_build:
     steps:
+      # TODO: remove the line below
+      # Before merging this PR we need to use an alternative develop branch to test the functionality, `develop-cache-test`
+
+      # On `develop-cache-test` we intentionally do NOT restore any ccache, so every run starts from scratch.
+      # Other branches restore the most recent cache produced on `develop-cache-test`.
+      - when:
+          condition:
+            not:
+              equal: ["develop-cache-test", << pipeline.git.branch >>]
+          steps:
+            - restore_cache:
+                keys:
+                  # Reuse cache produced on develop-cache-test (prefix match; restores the most recent).
+                  - v1-ccache-{{ arch }}-develop-cache-test-
+                  - v1-ccache-{{ arch }}-
```

Why do we need `v1-ccache-{{ arch }}-` here? You only ever save the cache for `develop-cache-test` so the second line will never match.

Did you mean something like this:

```suggestion
            - restore_cache:
                keys:
                  # Reuse the cache for current branch or fall back to the one from develop-cache-test if not available.
                  # NOTE: prefix match; restores the most recent
                  - v1-ccache-{{ arch }}-{{ .Branch }}-
                  - v1-ccache-{{ arch }}-develop-cache-test-
```

along with adjusting the save part to this?
```yaml
      - save_cache:
          key: v1-ccache-{{ arch }}-{{ .Branch }}-{{ .Revision }}
```

This would always start a new branch with cache from `develop`, but then keep using and updating the cache for the branch.

This would likely let us reuse more, though not sure I like it, because it does mean that the cache for branches is never recreated from scratch (unless we just nuke the whole cache by bumping the version) and is even reused for a completely new branch that just happens to share the same name as an old one.

---

Yeah, you are right it doesn't make sense to have `v1-ccache-{{ arch }}-` as fallback